tuning/data/data_processors.py

```diff
@@ -383,11 +380,128 @@ def _execute_data_handlers(
               {",".join([e.name for e in DataHandlerType])}'
         )
 
+    def _split_dataset(
+        self,
+        dataset_config: DataSetConfig,
+        loaded_dataset: Union[
+            Dataset, DatasetDict, IterableDataset, IterableDatasetDict
+        ],
+    ) -> Union[DatasetDict, IterableDatasetDict]:
+
+        seed = self.processor_config.seed
+        streaming = self.processor_config.streaming
+
+        train_split = dataset_config.split.get("train")
+        validation_split = dataset_config.split.get("validation")
+
+        if train_split is None or validation_split is None:
+            raise ValueError("Both 'train' and 'validation' splits must be specified!")
+
+        total_split = train_split + validation_split
+        if total_split > 1.0 or total_split == 0:
+            raise ValueError(f"Sum of splits must be in (0,1]. Got {total_split}")
+
+        # Todo : Add support for using multiple existing splits during spliting
+        # Problem : DatasetDict might have multiple splits already, which to use?
+        # One or more than one, or existing splitName other than train
+        # Currently Supports "train" by default
+        use_existing_splits = ["train"]  # For DatasetDict
+
+        if streaming:
```

Comment:
why do we need to check `if streaming` here? 
We can base all our checks on type of dataset

Reply:
Yeah we could have done it that way too but in my opinion it would be better to use streaming flag from data config as it would be better to decide intent of the  user from the config rather than from the type of Dataset the function receives.. altho both will work just fine but in my opinion I think we should let it be as it would be for better readability and design..